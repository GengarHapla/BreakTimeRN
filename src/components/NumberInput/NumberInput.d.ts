export interface INumberInputState {
    value: string;
}

export interface INumberInputProps {
    onChange: Function;
    placeholder: string;
}
