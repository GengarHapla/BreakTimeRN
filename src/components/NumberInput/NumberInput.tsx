import { StyleSheet, TextInput } from 'react-native';
import * as React from 'react';
import { INumberInputProps, INumberInputState } from './NumberInput.d';

class NumberInput extends React.Component<INumberInputProps, INumberInputState> {

    constructor(props: INumberInputProps) {
        super(props);

        this.state = {
            value: ''
        };
    }

    onValueChange(change: string) {
        const value = change.replace(/[^0-9]/g, '');
        this.setState({ value });
        this.props.onChange(value);
    }

    render() {
        return (
            <TextInput
                style={styles.textInput}
                keyboardType='numeric'
                underlineColorAndroid='transparent'
                onChangeText={(text: string) => this.onValueChange(text)}
                value={this.state.value}
                placeholder={this.props.placeholder}
            />
        );
    }
}

const styles = StyleSheet.create({
    textInput: {
        flex: 1,
        height: 40,
        padding: 4,
        borderColor: 'gray',
        borderWidth: 1,
        borderRadius: 4
    }
});

export default NumberInput;
