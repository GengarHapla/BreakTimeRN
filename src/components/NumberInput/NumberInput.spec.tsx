import React from 'react';
import NumberInput from './NumberInput';

import renderer from 'react-test-renderer';

test('renders correctly', () => {
    const tree = renderer.create(<NumberInput
            onChange={() => {}}
            placeholder={''}
            />).toJSON();
    expect(tree).toMatchSnapshot();
});
