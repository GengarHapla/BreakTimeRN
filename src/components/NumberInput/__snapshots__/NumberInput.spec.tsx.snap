// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<TextInput
  allowFontScaling={true}
  keyboardType="numeric"
  onChangeText={[Function]}
  placeholder=""
  style={
    Object {
      "borderColor": "gray",
      "borderRadius": 4,
      "borderWidth": 1,
      "flex": 1,
      "height": 40,
      "padding": 4,
    }
  }
  underlineColorAndroid="transparent"
  value=""
/>
`;
